lib/permissions/proposals/__tests__/listProposalCategoryPermissions.spec.ts

```
@@ -0,0 +1,121 @@
+import { InvalidInputError } from 'lib/utilities/errors';
+import { generateRole, generateUserAndSpace } from 'testing/setupDatabase';
+import { generateProposalCategory } from 'testing/utils/proposals';
+
+import { listProposalCategoryPermissions } from '../listProposalCategoryPermissions';
+import { upsertProposalCategoryPermission } from '../upsertProposalCategoryPermission';
+
+let space: Space;
+let user: User;
+let role: Role;
+
+beforeAll(async () => {
+  const generated = await generateUserAndSpace({
+    isAdmin: false
+  });
+  space = generated.space;
+  user = generated.user;
+  role = await generateRole({
+    createdBy: user.id,
+    spaceId: space.id,
+    assigneeUserIds: [user.id]
+  });
+});
+describe('listProposalCategoryPermissions', () => {
+  it('should return all assigned permissions for space members', async () => {
+    const proposalCategory = await generateProposalCategory({ spaceId: space.id });
+    const permissions = await Promise.all([
+      upsertProposalCategoryPermission({
+        assignee: { group: 'role', id: role.id },
+        permissionLevel: 'full_access',
+        proposalCategoryId: proposalCategory.id
+      }),
+      upsertProposalCategoryPermission({
+        assignee: { group: 'space', id: space.id },
+        permissionLevel: 'view',
+        proposalCategoryId: proposalCategory.id
+      }),
+      upsertProposalCategoryPermission({
+        assignee: { group: 'public' },
+        permissionLevel: 'view',
+        proposalCategoryId: proposalCategory.id
+      })
+    ]);
+
+    const foundPermissions = await listProposalCategoryPermissions({
+      resourceId: proposalCategory.id,
+      userId: user.id
+    });
+
+    expect(foundPermissions.length).toBe(permissions.length);
+    foundPermissions.forEach((foundPermission) => {
+      expect(permissions).toContainEqual(expect.objectContaining(foundPermission));
+    });
+  });
+
+  it('should return an empty list for non-space members', async () => {
```

since these are lib methods, i think they should throw an error if they are given garbage, just as you are throwing an error if the id is invalid.

---

We'll need to change this test :)